src/categorical_algebra/DataMigrations.jl

```diff
@@ -341,6 +355,84 @@ to recover the presentation from the datatype. Thus, this method for
 representable(::Type{T}, ob::Symbol) where T <: StructACSet =
   representable(T, Presentation(T), ob)
 
+
+"""
+The subobject classifier Ω in a presheaf topos is the presheaf that sends each 
+object A to the set of sieves on it (equivalently, the set of subobjects of the 
+representable presheaf for A). Counting subobjects gives us the *number* of A 
+parts; the hom data for f:A->B for subobject Aᵢ is determined via:
+
+Aᵢ ↪ A 
+↑    ↑ f*  
+PB⌝↪ B          (PB picks out a subobject of B, up to isomorphism.)
+
+(where A and B are the representables for objects A and B and f* is the unique 
+map from B into the A which sends the point of B to f applied to the point of A)
+
+Returns the classifier as well as a dictionary of subobjects corresponding to 
+the parts of the classifier.
+"""
+function subobject_classifier(T::Type, S::Presentation{ThSchema})
+  isempty(generators(S,:AttrType)) || error("Cannot compute Ω for schema with attributes")
+  obs    = nameof.(generators(S,:Ob))
+  reprs  = Dict(o=>representable(T, S,o) for o in obs)
+  subobs = Dict(o=>subobject_graph(reprs[o][1])[2] for o in obs)
+  Ω      = T()
+  for o in obs add_parts!(Ω, o, length(subobs[o])) end 
+
+  for (f, a, b) in homs(acset_schema(Ω))
+    BA = repr_map(Schema(S), reprs, f) 
+    Ω[f] = map(parts(Ω, a)) do pᵢ
+      Aᵢ = hom(subobs[a][pᵢ])
+      _, PB = force.(pullback(Aᵢ, BA))
+      return only(filter(parts(Ω, b)) do pⱼ
+        Bⱼ = hom(subobs[b][pⱼ])
+        any(σ ->  force(σ ⋅ Bⱼ) == PB, isomorphisms(dom(PB),dom(Bⱼ)))
+      end)
+    end
+  end
+  return Ω, subobs
+end
+
+"""
+Objects: Fᴳ(c) = C-Set(C × G, F)    where C is the representable c
+
+Given a map f: a->b, we compute that f(Aᵢ) = Bⱼ by constructing the following:
+          Aᵢ
+    A × G → F
+  f*↑ ↑ ↑ ↗ Bⱼ       find the hom Bⱼ that makes this commute
+    B × G 
+
+where f* is given by `repr_map`.
+"""
+function internal_hom(G::T, F::T, S::Presentation{ThSchema}) where T<:ACSet
+  obs    = nameof.(generators(S,:Ob))
+  reprs  = Dict(o=>representable(T, S,o) for o in obs)
+  prods  = Dict(o=>product(first(reprs[o]),G) for o in obs)
+  maps   = Dict(o=>homomorphisms(apex(prods[o]),F) for o in obs)
+  Fᴳ     = T()
+  for o in obs add_parts!(Fᴳ, o, length(maps[o])) end 
+  for (f, a, b) in homs(acset_schema(G))
+    BA = repr_map(Schema(S), reprs, f) 
+    π₁, π₂ = prods[b]
+    Fᴳ[f] = map(parts(Fᴳ, a)) do pᵢ
+      composite = force(universal(prods[a], Span(π₁⋅BA, π₂)) ⋅ maps[a][pᵢ])
+      findfirst(==(composite), maps[b])
+    end
+  end
+  return Fᴳ, homs
+end
+
+"""
+Any hom f: A->B induces a map from the representable of B to the repr of A
+"""
+function repr_map(S, reprs, f::Symbol)
```

Comment:
`repr` is already a Julia function with an unrelated meaning. Would call this `representable_map` for clarity.

Or better, now that I think of it: since this is just the action of the Yoneda embedding on morphisms and that seems generally useful, it would be nice to expose this through a public API, perhaps as a method for the generic function `yoneda` if that makes sense.